apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -180,7 +241,7 @@ public int hashCode() {
     }
 
     @Override
-    public void stop() {
-
+    public void stop() throws Exception {
+        server.removeNotificationListener(MBeanServerDelegate.DELEGATE_NAME, listener);
```

Comment:
probably better to avoid having `throws Exception` in the signature and handle it with a try/catch block here,
as expecting the method calling `stop()` to do it explicitly is not obvious as it is not part of the `LifecycleListener` interface definition.